packages/source-http/src/index.ts

```diff
@@ -29,11 +29,16 @@ export function getMetadataFromResponse(response: FetchLikeResponse): SourceMeta
 export class SourceHttp implements Source {
   type = 'http';
   url: URL;
+  headers?: Record<string,string>;
 
-  constructor(url: URL | string) {
+  constructor(url: URL | string, headers?:  Record<string, string>) {
     this.url = typeof url === 'string' ? SourceHttp.tryUrl(url) : url;
+    if (typeof headers !== 'undefined') {
+      this.headers = headers;
+    }
```

Comment:
We can just set this without checking as both `...this.headers` and `fetch({ headers: this.headers })` should work with undefined values

```suggestion
    this.headers = headers;
```